src/main/java/com/ays/user/controller/UserSelfController.java

```diff
@@ -0,0 +1,37 @@
+package com.ays.user.controller;
+
+import com.ays.common.model.dto.response.AysResponse;
+import com.ays.user.model.dto.request.UserSupportStatusUpdateRequest;
+import com.ays.user.service.UserSelfService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller class for managing the support status of a user.
+ */
+@RestController
+@RequestMapping("/api/v1/user/self")
+@RequiredArgsConstructor
+public class UserSelfController {
```

Comment:
Bu sınıfı dışarıdan çağırmadığımız için package-private yapabilriiz

Reply:
Gerekli değişikliği yaptım.